src/unit_tests/test_json_queue.c

```diff
@@ -0,0 +1,1070 @@
+
+    check_expected(formatted_msg);
+}
+
+extern FILE * __real_fopen(const char * __filename, const char * __modes);
+FILE * __wrap_fopen(const char * __filename, const char * __modes)
+{
+    if (test_mode) {
+        check_expected(__filename);
+        check_expected(__modes);
+
+        return mock_type(FILE *);
+    } else {
+        return __real_fopen(__filename, __modes);
+    }
+}
+
+extern int __real_fclose (FILE *__stream);
+int __wrap_fclose(FILE *__stream)
+{
+    if (test_mode) {
+        check_expected_ptr(__stream);
+
+        return 0;
+    } else {
+        return __real_fclose(__stream);
+    }
+}
+
+extern int __real_fseek(FILE *__stream, long int __off, int __whence);
+int __wrap_fseek(FILE *__stream, long int __off, int __whence)
+{
+    if (test_mode) {
+        return mock();
+    } else {
+        return __real_fseek(__stream, __off, __whence);
+    }
+}
+
+extern int __real_fstat (int __fd, struct stat *__buf);
+int __wrap_fstat(int __fd, struct stat *__buf)
+{
+    if (test_mode) {
+        return mock();
+    } else {
+        return __real_fstat(__fd, __buf);
+    }
+}
+
+void __wrap_fileno()
+{
+    return;
+}
+
+void __wrap_clearerr()
+{
+    return;
+}
+
+char * __wrap_fgets(char * __s, int __n, FILE * __stream)
+{
+    check_expected(__n);
+    check_expected_ptr(__stream);
+
+    strcpy(__s, mock_type(char *));
+
+    return mock_type(char *);
+}
+
+static int init_test_mode(void **state)
+{
+    (void) state;
+
+    test_mode = 1;
+
+    return 0;
+}
+
+static int end_test_mode(void **state)
+{
+    (void) state;
+
+    test_mode = 0;
+
+    return 0;
+}
+
+void test_jqueue_init(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+
+    jqueue_init(fileq);
+
+    assert_int_equal(fileq->last_change, 0);
+    assert_int_equal(fileq->year, 0);
+    assert_int_equal(fileq->day, 0);
+    assert_int_equal(fileq->flags, 0);
+    assert_null(fileq->mon[0]);
+    assert_null(fileq->file_name[0]);
+    assert_null(fileq->fp);
+
+    free(fileq);
+}
+
+void test_jqueue_open_fail_fopen(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+
+    /* fopen fail */
+
+    jqueue_init(fileq);
+
+    expect_string(__wrap_fopen, __filename, "/var/ossec/logs/alerts/alerts.json");
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, NULL);
+    expect_string(__wrap__merror, formatted_msg, "(1103): Could not open file '/var/ossec/logs/alerts/alerts.json' due to [(0)-(Success)].");
+
+    int ret = jqueue_open(fileq, 0);
+
+    assert_int_equal(ret, -1);
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_null(fileq->fp);
+
+    free(fileq);
+}
+
+void test_jqueue_open_fail_fseek(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+    FILE *fp;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+    os_calloc(1, sizeof(FILE), fp);
```

Comment:
Same as with test_jqueue_open_fail_fopen, if several variables need to be alloc'd they can be grouped into a structure and passed via `state`.

Reply:
Done!